src/scripting/lua_widget_attributes.cpp

```diff
@@ -0,0 +1,527 @@
+		if(i > n) {
+			throw std::invalid_argument("out of range");
+		}
+		return &tree_view_node->get_child_at(i - 1);
+	} else if(gui2::stacked_widget* stacked_widget = dynamic_cast<gui2::stacked_widget*>(&w)) {
+		int n = stacked_widget->get_layer_count();
+		if(i > n) {
+			throw std::invalid_argument("out of range");
+		}
+		return stacked_widget->get_layer_grid(i - 1);
+	}
+	return nullptr;
+}
+
+static gui2::widget* find_child_by_name(gui2::widget& w, const std::string& m)
+{
+    return w.find(m, false);
+}
+
+using tgetters = std::map<std::string, std::vector<std::function<bool(lua_State*, gui2::widget&)>>>;
+static tgetters getters;
+
+using tsetters = std::map<std::string, std::vector<std::function<bool(lua_State*, int, gui2::widget&)>>>;
+static tsetters setters;
+
+#define WIDGET_GETTER4(name, value_type, widgt_type, id) \
+/* use a class memeber for L to supress unused praemter wanring */ \
+struct BOOST_PP_CAT(getter_, id) { value_type do_it(widgt_type& w); lua_State* L; }; \
+struct BOOST_PP_CAT(getter_adder_, id) { \
+	BOOST_PP_CAT(getter_adder_, id) () \
+	{ \
+		utils::split_foreach(name, ',', 0, [](utils::string_view name_part){\
+			getters[std::string(name_part)].push_back([](lua_State* L, gui2::widget& w) { \
+				if(widgt_type* pw = dynamic_cast<widgt_type*>(&w)) { \
+					lua_push(L, BOOST_PP_CAT(getter_, id){L}.do_it(*pw)); \
+					return true; \
+				} \
+				return false; \
+			}); \
+		}); \
+	} \
+}; \
+static BOOST_PP_CAT(getter_adder_, id) BOOST_PP_CAT(getter_adder_instance_, id) ; \
+value_type BOOST_PP_CAT(getter_, id)::do_it(widgt_type& w)
+
+
+#define WIDGET_SETTER4(name, value_type, widgt_type, id) \
+struct BOOST_PP_CAT(setter_, id) { void do_it(widgt_type& w, const value_type& value); lua_State* L; }; \
+struct BOOST_PP_CAT(setter_adder_, id) { \
+	BOOST_PP_CAT(setter_adder_, id) ()\
+	{ \
+		utils::split_foreach(name, ',', 0, [](utils::string_view name_part){\
+			setters[std::string(name_part)].push_back([](lua_State* L, int idx, gui2::widget& w) { \
+				if(widgt_type* pw = dynamic_cast<widgt_type*>(&w)) { \
+					BOOST_PP_CAT(setter_, id){L}.do_it(*pw, lua_check<value_type>(L, idx)); \
+					return true; \
+				} \
+				return false; \
+			}); \
+		}); \
+	} \
+}; \
+static BOOST_PP_CAT(setter_adder_, id) BOOST_PP_CAT(setter_adder_instance_, id); \
+void BOOST_PP_CAT(setter_, id)::do_it(widgt_type& w, const value_type& value)
+
+
+/**
+ * @param name: string  comma seperated list
+ * @param type: the type of the attribute, for exmaple int or std::string
+ * @param widgt_type: the type of the widget, for exmaple gui2::listbox
+ */
+
+#define WIDGET_GETTER(name, value_type, widgt_type) WIDGET_GETTER4(name, value_type, widgt_type, __LINE__)
+
+#define WIDGET_SETTER(name, value_type, widgt_type) WIDGET_SETTER4(name, value_type, widgt_type, __LINE__)
+
+
+/// CLASSIC
+
+WIDGET_GETTER("value_compat,selected_index", int, gui2::listbox)
+{
+	return w.get_selected_row() + 1;
+}
+
+/* idea
```

Comment:
This seems like a good idea! Was there some technical reason why it didn't work?

Reply:
Hmm not really i t can be one with a bit i just wans't sure whwther its useful, and i prefer to rather not to add something in such cases.

Reply:
Hmm. Well, it's pretty easy to do it in Lua with the existing API anyway, so probably not that important. You'd just do `widget[widget.selected_index]`, right?

Reply:
Yes